crates/indexer/src/error.rs

```diff
@@ -0,0 +1,7 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum IndexerError {
+    #[error("connection failed by: {0}, error: {1}")]
+    ConnectionError(String, anyhow::Error),
+}
```

Comment:
Why not change the rpc client crate to return this error instead of `anyhow::Error`. This way you don't need to call `map_err` at every call site.

Reply:
updated